### DISCUSSION TITLE
2021年8月需求墙

### DISCUSSION THREAD

**Discussion by backrunner** [pwp-app/pixiviz#22] [* 公告区 (Announcements)]
# 2021年8月需求墙

这里是2021年7-8月中用户反馈的、已经评估过可以进行开发的需求。

**Comment by backrunner** [comment]
## 首页搜索可以根据名字搜索画师

已跟随 v1.24.0 上线

- [x] 首页搜索修改，可以通过下拉选择搜索画师。
- [x] 独立的画师搜索页面
- [x] 公共画师页卡
- [x] 对应的API

**Comment by backrunner** [comment]
## 社会化分享

已跟随 v1.24.x 上线

- [x] Open Graph Tags
- [x] 图片详情页内的独立的社会化分享按钮 (Independent social share button in picture detail page)

**Comment by backrunner** [comment]
## 服务器选择面板（Priority: High）

~~在站点扩容后，计划在【首页Banner/设置】中增加服务器线路选择Dialog。~~

修改成全自动扫描状态 + 切换线路，跟随 v1.24.5 上线

**Comment by backrunner** [comment]
## 使用 Hash 算法替代 LoadMap（Priority：Very High）

已跟随 v1.24.3 上线

Thanks @Rorical.
使用 Hash 算法在保持负载均衡权重的情况下，确定图片服务器的分配。

- [x] 剔除 LoadMap，使用 Hash 算法替代。

**Reply by Rorical** [reply]
我觉得用哈希进行图片服务器的分配也是个办法

**Reply by backrunner** [reply]
> 我觉得用哈希进行图片服务器的分配也是个办法

好主意，这样可以彻底丢掉loadMap

**Comment by backrunner** [comment]
## 使用指南迁移(Priority：Low, Delayed)

使用指南需要迁移到以 Fragy 为基础的新站点。

- [x] Fragy 站点搭建
- [x] 配置文件更新

**Comment by backrunner** [comment]
## Banner 展开的新用户引导（Priority：Low）

- [ ] 面向新用户添加 Overlay，引导用户双击 Banner。

**Comment by backrunner** [comment]
## 维护中 Overlay（Priority：Low）

当远程配置包含`"maintaining": true`时，对所有进入页面的用户展示维护中的浮层。

- [x] 维护中浮层
- [x] 远程自定义文案展示

**Comment by backrunner** [comment]
## Rank 加载失败重试按钮（Priority：High）

在因为网络原因，拉取 Rank 数据失败后，给用户展示重试按钮，节约刷新页面的成本。

- [x] 添加重试按钮

已上线

**Comment by backrunner** [comment]
## 页面底部弱网提示（Priroity: High）

在检测到用户弱网或可能弱网的情况下，在页面底部弹出提示。

已上线
